src/db/migrations/20190314023509_connect_personhist_saldohistory.js

```diff
@@ -37,22 +37,29 @@ export const up = async (knex) => {
 			saldoIndex++;
 		}
 
-		/* The next saldo event should match the deposit event. Its timestamp should be 0s or 1s later. Just to be sure,
-		 * this checks if its properties actually match the deposit event. */
-		const saldoEvent = unconnectedSaldoEvents[saldoIndex];
-		if (
-			(Date.parse(saldoEvent.time) === depositTime || Date.parse(saldoEvent.time) === depositTime + 1000) &&
-			saldoEvent.userid === depositEvent.userid1
-		) {
-			/* Inserting saldo event references to deposit event objects. */
-			depositEvent.saldhistid = saldoEvent.saldhistid;
+		// Consider the next 10 saldo events just to be safe with ordering.
+		// In production data there are cases where multiple saldo events are at the same time,
+		// with the deposit event being sorted _after_ a saldo event pertaining to a user with a lower id.
+		for (let i = 0; i < 10; i++) {
```

Comment:
There are cases where there are multiple unmapped saldo history events in the same second, and if one of them is by a user that has a lower id than the currently considered deposit event's user's id, there's a mismatch. This could probably be lowered down to 5 without causing issues.

Reply:
Migrations are a one-and-done anyways, so taking a bit longer is no big deal.